src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -360,9 +360,70 @@ Convert Convert::deserialize(std::istream& stream) {
     read(stream, c.tensor);
     return c;
 }
+
+std::size_t Gather::hash() const {
+    std::size_t seed = w.get_hash() + 0x9e3779b9;
+    seed ^= t.get_element_type().hash() + 0x9e3779b9;
+    for (const auto& dim : t.get_shape()) {
+        seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
+    }
+    seed ^= std::hash<const void*>()(orig_lut_ptr) + 0x9e3779b9;
+    seed ^= dst_type.hash() + 0x9e3779b9;
+    for (const auto& dim : dst_shape) {
+        seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
+    }
+    return seed;
+}
+
+bool Gather::operator==(const Gather& other) const {
+    return (w == other.w && t.get_element_type() == other.t.get_element_type() &&
+            t.get_shape() == other.t.get_shape() && orig_lut_ptr == other.orig_lut_ptr && dst_type == other.dst_type &&
+            dst_shape == other.dst_shape);
+}
+
+ov::Tensor Gather::eval() const {
+    auto ttype = t.get_element_type();
+    NPUW_ASSERT(ttype == ov::element::f8e4m3 || ttype == ov::element::f8e5m2 || ttype == ov::element::f8e8m0);
+    ov::Tensor dst(dst_type, dst_shape);
+    const auto& gti = ov::get_tensor_impl;
+    ov::npuw::util::gather_nf4(gti(t), gti(w.eval()), gti(dst));
+    return dst;
+}
+
+LazyTensor::Meta Gather::eval_meta() const {
+    return {dst_shape, dst_type};
+}
+
+void Gather::read_weight(const ov::npuw::s11n::WeightsContext& ctx) {
+    w.read_weight(ctx);
+}
+
+void Gather::detach() {
+    w.detach();
+}
+
+void Gather::serialize(std::ostream& stream) const {
+    using namespace ov::npuw::s11n;
+    write(stream, dst_type.to_string());
+    write(stream, dst_shape);
+    write(stream, w);
+    write(stream, t);
+}
+
+Gather Gather::deserialize(std::istream& stream) {
+    using namespace ov::npuw::s11n;
+    Gather g;
+    std::string type_str;
+    read(stream, type_str);
+    g.dst_type = ov::element::Type(type_str);
+    read(stream, g.dst_shape);
+    read(stream, g.w);
+    read(stream, g.t);
+    return g;
+}
```

Comment:
So should you bump the serialization tag here or not?
Previously you didn't write lazy tensors of this type but now you do